tests/test_core_network.py

```diff
@@ -2656,8 +2657,12 @@ def test_generate_validation_report_with_pt2matsim_network(network_object_from_t
                 'bike': {'problem_nodes': {'dead_ends': [], 'unreachable_node': []},
                          'number_of_connected_subgraphs': 0}},
             'link_attributes': {
-                'links_over_1km_length': {'number_of': 0, 'percentage': 0.0, 'link_ids': []},
-                'zero_attributes': {}}},
+                'links_over_1000_length': {'number_of': 0, 'percentage': 0.0, 'link_ids': []},
+                'zero_attributes': {},
+                'negative_attributes': {},
+                'infinite_attributes': {},
+                'fractional_attributes': {}
```

Comment:
We could be a bit more fine-grained in our tests here. For example, I can imagine a test that uses a network with a zero attribute. We expect that to be reflected in the report. The assertion would be much narrower - grab the `link_attributes` sub dict and assert that it looks how you expect, but ignore the rest of the report. Or go even further and assert only on `report['graph']['link_attributes']['zero_attributes']`.

Reply:
yeah there are a few of these old blanket tests hanging about, got new and better tests for the link attribute checks, I should just take the opportunity to chop them up and make them into proper tests